### PR TITLE
fix: Verbose curl only in debug mode

### PR DESCRIPTION
**Pull request by rhcarvalho**
Setting verbose output for curl produces annoying output like
'* Closing connection 0'

This change makes sentry-cli less verbose in its normal usage and keeps
the verbose mode of curl and header dumping when --log-level=debug.

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -1308,8 +1308,19 @@ fn handle_req<W: Write>(
         handle.progress(true)?;
     }
 
-    // enable verbose mode
-    handle.verbose(true)?;
+    if Config::current().get_log_level() == log::LevelFilter::Debug {
```
Comment (jan-auer) [inline src/api.rs]: There is also a `trace` log level.

```suggestion
    if Config::current().get_log_level() >= log::LevelFilter::Debug {
```

**Comment by kamilogorek** [comment]
I pointed out where in curl it has been fixed here https://github.com/getsentry/sentry-cli/pull/657#discussion_r378691354
Your call which one should we merge. I don't have enough insight yet.

**Comment by kamilogorek** [comment]
Fixed in https://github.com/getsentry/sentry-cli/pull/657

**Comment by rhcarvalho** [comment]
I wonder if this is useful anyway to reduce output under normal operation? Haven't noticed yet what the debug output adds, though.

**Comment by kamilogorek** [comment]
I haven't seen any yet tbh. Will keep that in mind if I'll stumble upon one

**Pull request closed without merging**